Varargs method for Entity.add(Component)

Alexander9211 opened the pull request:
Simply added a vararg utility method for Entity.add(Component), for code readability.

saltares commented:
Can you please add unit tests?

saltares commented on ashley/src/com/badlogic/ashley/core/Entity.java:
```diff
@@ -75,6 +75,18 @@ public Entity add (Component component) {
 	}
+	
+	/**
+	 * Adds all given {@link Component}s to this Entity. See {@link Entity#add(Component)}.
```
Could you add that this will allocate memory and generate work for the GC? 

http://stackoverflow.com/questions/7452513/java-varargs-allocates-array-objects

Alexander9211 commented:
I'm not entirely sure if this is what you wanted, but I added a quick test for the vararg method and fixed the description. I didn't know about memory allocation for varargs! Sorry for taking my time but I'm pretty new to using GitHub and it took me a while to setup stuffs on my IDE.

If there's anything more I can do just tell me, and sorry if I misunderstood :)

antag99 commented:
I don't understand why this is necessary - IMO it only bloats the API. You could write your own class for creating entities, if chaining calls to `Entity.add(Component)` isn't sufficient. E.g.
```java
engine.createEntity()
    .add(new Position(5f, 5f))
    .add(new Size(5f, 5f));
```
looks pretty enough to me.

saltares commented:
@Alexander9211 
> I'm not entirely sure if this is what you wanted, but I added a quick test for the vararg method and fixed the description

I meant add a proper unit tests to the [Entity unit tests](https://github.com/libgdx/ashley/blob/master/ashley/tests/com/badlogic/ashley/core/EntityTests.java).

@antag99 is right, you can already chain calls to `add()`.

Alexander9211 commented:
Sorry I'm still unexperienced so I didn't know about JUnit. As for the chain / vararg dispute, it just came to my mind after writing the 10th "add" in a chain in my game's builder (but maybe I'm just overdoing with components), I didn't think it would be that much of a deal. Sorry for wasting your time, have a nice day.
